17/03/15 23:21:36 WARN SparkContext: Support for Java 7 is deprecated as of Spark 2.0.0
17/03/15 23:21:36 DEBUG MutableMetricsFactory: field org.apache.hadoop.metrics2.lib.MutableRate org.apache.hadoop.security.UserGroupInformation$UgiMetrics.loginSuccess with annotation @org.apache.hadoop.metrics2.annotation.Metric(about=, value=[Rate of successful kerberos logins and latency (milliseconds)], valueName=Time, always=false, type=DEFAULT, sampleName=Ops)
17/03/15 23:21:36 DEBUG MutableMetricsFactory: field org.apache.hadoop.metrics2.lib.MutableRate org.apache.hadoop.security.UserGroupInformation$UgiMetrics.loginFailure with annotation @org.apache.hadoop.metrics2.annotation.Metric(about=, value=[Rate of failed kerberos logins and latency (milliseconds)], valueName=Time, always=false, type=DEFAULT, sampleName=Ops)
17/03/15 23:21:36 DEBUG MutableMetricsFactory: field org.apache.hadoop.metrics2.lib.MutableRate org.apache.hadoop.security.UserGroupInformation$UgiMetrics.getGroups with annotation @org.apache.hadoop.metrics2.annotation.Metric(about=, value=[GetGroups], valueName=Time, always=false, type=DEFAULT, sampleName=Ops)
17/03/15 23:21:36 DEBUG MetricsSystemImpl: UgiMetrics, User and group related metrics
17/03/15 23:21:36 DEBUG Shell: setsid exited with exit code 0
17/03/15 23:21:36 DEBUG Groups:  Creating new Groups object
17/03/15 23:21:36 DEBUG NativeCodeLoader: Trying to load the custom-built native-hadoop library...
17/03/15 23:21:36 DEBUG NativeCodeLoader: Failed to load native-hadoop with error: java.lang.UnsatisfiedLinkError: no hadoop in java.library.path
17/03/15 23:21:36 DEBUG NativeCodeLoader: java.library.path=/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
17/03/15 23:21:36 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
17/03/15 23:21:36 DEBUG PerformanceAdvisory: Falling back to shell based
17/03/15 23:21:36 DEBUG JniBasedUnixGroupsMappingWithFallback: Group mapping impl=org.apache.hadoop.security.ShellBasedUnixGroupsMapping
17/03/15 23:21:36 DEBUG Groups: Group mapping impl=org.apache.hadoop.security.JniBasedUnixGroupsMappingWithFallback; cacheTimeout=300000; warningDeltaMs=5000
17/03/15 23:21:36 DEBUG UserGroupInformation: hadoop login
17/03/15 23:21:36 DEBUG UserGroupInformation: hadoop login commit
17/03/15 23:21:36 DEBUG UserGroupInformation: using local user:UnixPrincipal: hadoop
17/03/15 23:21:36 DEBUG UserGroupInformation: Using user: "UnixPrincipal: hadoop" with name hadoop
17/03/15 23:21:36 DEBUG UserGroupInformation: User entry: "hadoop"
17/03/15 23:21:36 DEBUG UserGroupInformation: UGI loginUser:hadoop (auth:SIMPLE)
17/03/15 23:21:36 WARN SparkConf: Detected deprecated memory fraction settings: [spark.storage.memoryFraction]. As of Spark 1.6, execution and storage memory management are unified. All memory fractions used in the old model are now deprecated and no longer read. If you wish to use the old memory management, you may explicitly enable `spark.memory.useLegacyMode` (not recommended).
17/03/15 23:21:36 WARN SparkConf: 
SPARK_CLASSPATH was detected (set to '/home/hadoop/bryantchang/platforms/spark2.0/lib/mysql-connector-java-5.1.40-bin.jar:').
This is deprecated in Spark 1.0+.

Please instead use:
 - ./spark-submit with --driver-class-path to augment the driver classpath
 - spark.executor.extraClassPath to augment the executor classpath
        
17/03/15 23:21:36 WARN SparkConf: Setting 'spark.executor.extraClassPath' to '/home/hadoop/bryantchang/platforms/spark2.0/lib/mysql-connector-java-5.1.40-bin.jar:' as a work-around.
17/03/15 23:21:36 WARN SparkConf: Setting 'spark.driver.extraClassPath' to '/home/hadoop/bryantchang/platforms/spark2.0/lib/mysql-connector-java-5.1.40-bin.jar:' as a work-around.
17/03/15 23:21:36 DEBUG InternalLoggerFactory: Using SLF4J as the default logging framework
17/03/15 23:21:36 DEBUG PlatformDependent0: java.nio.Buffer.address: available
17/03/15 23:21:36 DEBUG PlatformDependent0: sun.misc.Unsafe.theUnsafe: available
17/03/15 23:21:36 DEBUG PlatformDependent0: sun.misc.Unsafe.copyMemory: available
17/03/15 23:21:36 DEBUG PlatformDependent0: direct buffer constructor: available
17/03/15 23:21:36 DEBUG PlatformDependent0: java.nio.Bits.unaligned: available, true
17/03/15 23:21:36 DEBUG PlatformDependent0: java.nio.DirectByteBuffer.<init>(long, int): available
17/03/15 23:21:36 DEBUG Cleaner0: java.nio.ByteBuffer.cleaner(): available
17/03/15 23:21:36 DEBUG PlatformDependent: Java version: 7
17/03/15 23:21:36 DEBUG PlatformDependent: -Dio.netty.noUnsafe: false
17/03/15 23:21:36 DEBUG PlatformDependent: sun.misc.Unsafe: available
17/03/15 23:21:36 DEBUG PlatformDependent: -Dio.netty.noJavassist: false
17/03/15 23:21:36 DEBUG PlatformDependent: Javassist: available
17/03/15 23:21:36 DEBUG PlatformDependent: -Dio.netty.tmpdir: /tmp (java.io.tmpdir)
17/03/15 23:21:36 DEBUG PlatformDependent: -Dio.netty.bitMode: 64 (sun.arch.data.model)
17/03/15 23:21:36 DEBUG PlatformDependent: -Dio.netty.noPreferDirect: false
17/03/15 23:21:36 DEBUG PlatformDependent: io.netty.maxDirectMemory: 954728448 bytes
17/03/15 23:21:36 DEBUG JavassistTypeParameterMatcherGenerator: Generated: io.netty.util.internal.__matchers__.org.apache.spark.network.protocol.MessageMatcher
17/03/15 23:21:36 DEBUG JavassistTypeParameterMatcherGenerator: Generated: io.netty.util.internal.__matchers__.io.netty.buffer.ByteBufMatcher
17/03/15 23:21:36 DEBUG MultithreadEventLoopGroup: -Dio.netty.eventLoopThreads: 8
17/03/15 23:21:36 DEBUG NioEventLoop: -Dio.netty.noKeySetOptimization: false
17/03/15 23:21:36 DEBUG NioEventLoop: -Dio.netty.selectorAutoRebuildThreshold: 512
17/03/15 23:21:36 DEBUG PlatformDependent: org.jctools-core.MpscChunkedArrayQueue: available
17/03/15 23:21:36 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.numHeapArenas: 8
17/03/15 23:21:36 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.numDirectArenas: 8
17/03/15 23:21:36 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.pageSize: 8192
17/03/15 23:21:36 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.maxOrder: 11
17/03/15 23:21:36 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.chunkSize: 16777216
17/03/15 23:21:36 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.tinyCacheSize: 512
17/03/15 23:21:36 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.smallCacheSize: 256
17/03/15 23:21:36 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.normalCacheSize: 64
17/03/15 23:21:36 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.maxCachedBufferCapacity: 32768
17/03/15 23:21:36 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.cacheTrimInterval: 8192
17/03/15 23:21:36 DEBUG ThreadLocalRandom: -Dio.netty.initialSeedUniquifier: 0xd6960422d40a5877 (took 0 ms)
17/03/15 23:21:36 DEBUG ByteBufUtil: -Dio.netty.allocator.type: unpooled
17/03/15 23:21:36 DEBUG ByteBufUtil: -Dio.netty.threadLocalDirectBufferSize: 65536
17/03/15 23:21:36 DEBUG ByteBufUtil: -Dio.netty.maxThreadLocalCharBufferSize: 16384
17/03/15 23:21:36 DEBUG NetUtil: Loopback interface: lo (lo, 0:0:0:0:0:0:0:1%1)
17/03/15 23:21:36 DEBUG NetUtil: /proc/sys/net/core/somaxconn: 128
17/03/15 23:21:37 DEBUG AbstractByteBuf: -Dio.netty.buffer.bytebuf.checkAccessible: true
17/03/15 23:21:37 DEBUG ResourceLeakDetector: -Dio.netty.leakDetection.level: simple
17/03/15 23:21:37 DEBUG ResourceLeakDetector: -Dio.netty.leakDetection.maxRecords: 4
17/03/15 23:21:37 DEBUG ResourceLeakDetectorFactory: Loaded default ResourceLeakDetector: io.netty.util.ResourceLeakDetector@6b2685fb
17/03/15 23:21:37 DEBUG Recycler: -Dio.netty.recycler.maxCapacity.default: 32768
17/03/15 23:21:37 DEBUG Recycler: -Dio.netty.recycler.maxSharedCapacityFactor: 2
17/03/15 23:21:37 DEBUG Recycler: -Dio.netty.recycler.linkCapacity: 16
17/03/15 23:21:37 DEBUG Recycler: -Dio.netty.recycler.ratio: 8
17/03/15 23:21:37 DEBUG BlockReaderLocal: dfs.client.use.legacy.blockreader.local = false
17/03/15 23:21:37 DEBUG BlockReaderLocal: dfs.client.read.shortcircuit = false
17/03/15 23:21:37 DEBUG BlockReaderLocal: dfs.client.domain.socket.data.traffic = false
17/03/15 23:21:37 DEBUG BlockReaderLocal: dfs.domain.socket.path = 
17/03/15 23:21:37 DEBUG RetryUtils: multipleLinearRandomRetry = null
17/03/15 23:21:37 DEBUG Server: rpcKind=RPC_PROTOCOL_BUFFER, rpcRequestWrapperClass=class org.apache.hadoop.ipc.ProtobufRpcEngine$RpcRequestWrapper, rpcInvoker=org.apache.hadoop.ipc.ProtobufRpcEngine$Server$ProtoBufRpcInvoker@226c4923
17/03/15 23:21:37 DEBUG Client: getting client out of cache: org.apache.hadoop.ipc.Client@1a6f3919
17/03/15 23:21:37 DEBUG PerformanceAdvisory: Both short-circuit local reads and UNIX domain socket are disabled.
17/03/15 23:21:37 DEBUG DataTransferSaslUtil: DataTransferProtocol not using SaslPropertiesResolver, no QOP found in configuration for dfs.data.transfer.protection
17/03/15 23:21:37 DEBUG Client: The ping interval is 60000 ms.
17/03/15 23:21:37 DEBUG Client: Connecting to spark1/172.21.15.90:9000
17/03/15 23:21:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: starting, having connections 1
17/03/15 23:21:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #0
17/03/15 23:21:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #0
17/03/15 23:21:37 DEBUG ProtobufRpcEngine: Call: getFileInfo took 24ms
17/03/15 23:21:37 DEBUG DFSClient: /eventLogs/app-20170315232137-0002.lz4.inprogress: masked=rw-r--r--
17/03/15 23:21:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #1
17/03/15 23:21:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #1
17/03/15 23:21:37 DEBUG ProtobufRpcEngine: Call: create took 16ms
17/03/15 23:21:37 DEBUG DFSClient: computePacketChunkSize: src=/eventLogs/app-20170315232137-0002.lz4.inprogress, chunkSize=516, chunksPerPacket=126, packetSize=65016
17/03/15 23:21:37 DEBUG LeaseRenewer: Lease renewer daemon for [DFSClient_NONMAPREDUCE_-308821755_1] with renew id 1 started
17/03/15 23:21:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #2
17/03/15 23:21:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #2
17/03/15 23:21:37 DEBUG ProtobufRpcEngine: Call: setPermission took 7ms
17/03/15 23:21:37 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=0 lastFlushOffset=0 createNewBlock=false
17/03/15 23:21:37 DEBUG DFSClient: Waiting for ack for: -1
START load
17/03/15 23:21:37 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=0 lastFlushOffset=0 createNewBlock=false
17/03/15 23:21:37 DEBUG DFSClient: Waiting for ack for: -1
START training
17/03/15 23:21:38 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #3
17/03/15 23:21:38 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #3
17/03/15 23:21:38 DEBUG ProtobufRpcEngine: Call: getFileInfo took 1ms
17/03/15 23:21:38 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #4
17/03/15 23:21:38 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #4
17/03/15 23:21:38 DEBUG ProtobufRpcEngine: Call: getListing took 1ms
17/03/15 23:21:38 DEBUG FileInputFormat: Time taken to get FileStatuses: 16
17/03/15 23:21:38 INFO FileInputFormat: Total input paths to process : 10
17/03/15 23:21:38 DEBUG FileInputFormat: Total # of splits generated by getSplits: 10, TimeTaken: 20
17/03/15 23:21:38 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=0 lastFlushOffset=0 createNewBlock=false
17/03/15 23:21:38 DEBUG DFSClient: Waiting for ack for: -1
17/03/15 23:21:38 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=0, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=0
17/03/15 23:21:38 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=6244 lastFlushOffset=0 createNewBlock=false
17/03/15 23:21:38 DEBUG DFSClient: Queued packet 0
17/03/15 23:21:38 DEBUG DFSClient: Waiting for ack for: 0
17/03/15 23:21:38 DEBUG DFSClient: Allocating new block
17/03/15 23:21:38 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #5
17/03/15 23:21:38 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #5
17/03/15 23:21:38 DEBUG ProtobufRpcEngine: Call: addBlock took 7ms
17/03/15 23:21:38 DEBUG DFSClient: pipeline = DatanodeInfoWithStorage[172.21.15.173:50010,DS-bcd3842f-7acc-473a-9a24-360950418375,DISK]
17/03/15 23:21:38 DEBUG DFSClient: Connecting to datanode 172.21.15.173:50010
17/03/15 23:21:38 DEBUG DFSClient: Send buf size 124928
17/03/15 23:21:38 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #6
17/03/15 23:21:38 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #6
17/03/15 23:21:38 DEBUG ProtobufRpcEngine: Call: getServerDefaults took 0ms
17/03/15 23:21:38 DEBUG SaslDataTransferClient: SASL client skipping handshake in unsecured configuration for addr = /172.21.15.173, datanodeId = DatanodeInfoWithStorage[172.21.15.173:50010,DS-bcd3842f-7acc-473a-9a24-360950418375,DISK]
17/03/15 23:21:38 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809232_68409 sending packet packet seqno: 0 offsetInBlock: 0 lastPacketInBlock: false lastByteOffsetInBlock: 6244
17/03/15 23:21:38 DEBUG DFSClient: DFSClient seqno: 0 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0
17/03/15 23:21:38 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #7
17/03/15 23:21:38 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #7
17/03/15 23:21:38 DEBUG ProtobufRpcEngine: Call: fsync took 9ms
17/03/15 23:21:38 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=1, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=6144
17/03/15 23:21:38 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=6244 lastFlushOffset=6244 createNewBlock=false
17/03/15 23:21:38 DEBUG DFSClient: Waiting for ack for: 0
17/03/15 23:21:41 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=2, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=6144
17/03/15 23:21:41 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=6244 lastFlushOffset=6244 createNewBlock=false
17/03/15 23:21:41 DEBUG DFSClient: Waiting for ack for: 0
17/03/15 23:21:41 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=3, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=6144
17/03/15 23:21:41 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=6244 lastFlushOffset=6244 createNewBlock=false
17/03/15 23:21:41 DEBUG DFSClient: Waiting for ack for: 0
17/03/15 23:21:41 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=4, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=6144
17/03/15 23:21:41 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=6244 lastFlushOffset=6244 createNewBlock=false
17/03/15 23:21:41 DEBUG DFSClient: Waiting for ack for: 0
17/03/15 23:21:41 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=5, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=6144
17/03/15 23:21:41 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=6244 lastFlushOffset=6244 createNewBlock=false
17/03/15 23:21:41 DEBUG DFSClient: Waiting for ack for: 0
17/03/15 23:21:41 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=6, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=6144
17/03/15 23:21:41 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=6244 lastFlushOffset=6244 createNewBlock=false
17/03/15 23:21:41 DEBUG DFSClient: Waiting for ack for: 0
17/03/15 23:21:41 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=7, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=6144
17/03/15 23:21:41 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=6244 lastFlushOffset=6244 createNewBlock=false
17/03/15 23:21:41 DEBUG DFSClient: Waiting for ack for: 0
[Stage 2:=====>                                                    (1 + 0) / 10][Stage 2:=====>                                                    (1 + 4) / 10]17/03/15 23:21:48 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: closed
17/03/15 23:21:48 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: stopped, remaining connections 0
[Stage 2:=============================>                            (5 + 4) / 10][Stage 2:====================================================>     (9 + 1) / 10]17/03/15 23:21:56 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=8, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=6144
17/03/15 23:21:56 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=13825 lastFlushOffset=6244 createNewBlock=false
17/03/15 23:21:56 DEBUG DFSClient: Queued packet 8
17/03/15 23:21:56 DEBUG DFSClient: Waiting for ack for: 8
17/03/15 23:21:56 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809232_68409 sending packet packet seqno: 8 offsetInBlock: 6144 lastPacketInBlock: false lastByteOffsetInBlock: 13825
17/03/15 23:21:56 DEBUG DFSClient: DFSClient seqno: 8 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0
                                                                                17/03/15 23:21:57 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=9, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=13824
17/03/15 23:21:57 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=17888 lastFlushOffset=13825 createNewBlock=false
17/03/15 23:21:57 DEBUG DFSClient: Queued packet 9
17/03/15 23:21:57 DEBUG DFSClient: Waiting for ack for: 9
17/03/15 23:21:57 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809232_68409 sending packet packet seqno: 9 offsetInBlock: 13824 lastPacketInBlock: false lastByteOffsetInBlock: 17888
17/03/15 23:21:57 DEBUG DFSClient: DFSClient seqno: 9 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0
17/03/15 23:21:57 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=10, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=17408
17/03/15 23:21:57 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=17888 lastFlushOffset=17888 createNewBlock=false
17/03/15 23:21:57 DEBUG DFSClient: Waiting for ack for: 9
17/03/15 23:21:57 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=11, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=17408
17/03/15 23:21:57 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=17888 lastFlushOffset=17888 createNewBlock=false
17/03/15 23:21:57 DEBUG DFSClient: Waiting for ack for: 9
[Stage 4:>                                                         (0 + 0) / 10][Stage 4:>                                                         (0 + 4) / 10]17/03/15 23:22:07 DEBUG Client: The ping interval is 60000 ms.
17/03/15 23:22:07 DEBUG Client: Connecting to spark1/172.21.15.90:9000
17/03/15 23:22:07 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: starting, having connections 1
17/03/15 23:22:07 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #8
17/03/15 23:22:07 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #8
17/03/15 23:22:07 DEBUG ProtobufRpcEngine: Call: renewLease took 9ms
17/03/15 23:22:07 DEBUG LeaseRenewer: Lease renewed for client DFSClient_NONMAPREDUCE_-308821755_1
17/03/15 23:22:07 DEBUG LeaseRenewer: Lease renewer daemon for [DFSClient_NONMAPREDUCE_-308821755_1] with renew id 1 executed
17/03/15 23:22:17 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: closed
17/03/15 23:22:17 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: stopped, remaining connections 0
[Stage 4:=======================>                                  (4 + 4) / 10]17/03/15 23:22:37 DEBUG Client: The ping interval is 60000 ms.
17/03/15 23:22:37 DEBUG Client: Connecting to spark1/172.21.15.90:9000
17/03/15 23:22:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: starting, having connections 1
17/03/15 23:22:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #9
17/03/15 23:22:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #9
17/03/15 23:22:37 DEBUG ProtobufRpcEngine: Call: renewLease took 7ms
17/03/15 23:22:37 DEBUG LeaseRenewer: Lease renewed for client DFSClient_NONMAPREDUCE_-308821755_1
17/03/15 23:22:37 DEBUG LeaseRenewer: Lease renewer daemon for [DFSClient_NONMAPREDUCE_-308821755_1] with renew id 1 executed
[Stage 4:=============================>                            (5 + 4) / 10][Stage 4:========================================>                 (7 + 3) / 10][Stage 4:==============================================>           (8 + 2) / 10]17/03/15 23:22:47 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: closed
17/03/15 23:22:47 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: stopped, remaining connections 0
[Stage 4:====================================================>     (9 + 1) / 10]17/03/15 23:22:55 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=12, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=17408
17/03/15 23:22:55 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=22583 lastFlushOffset=17888 createNewBlock=false
17/03/15 23:22:55 DEBUG DFSClient: Queued packet 12
17/03/15 23:22:55 DEBUG DFSClient: Waiting for ack for: 12
17/03/15 23:22:55 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809232_68409 sending packet packet seqno: 12 offsetInBlock: 17408 lastPacketInBlock: false lastByteOffsetInBlock: 22583
17/03/15 23:22:55 WARN DFSClient: Slow ReadProcessor read fields took 57863ms (threshold=30000ms); ack: seqno: 12 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0, targets: [DatanodeInfoWithStorage[172.21.15.173:50010,DS-bcd3842f-7acc-473a-9a24-360950418375,DISK]]
[Stage 5:>                                                          (0 + 2) / 2]17/03/15 23:22:56 WARN TaskSetManager: Lost task 0.0 in stage 5.0 (TID 24, 172.21.15.173, executor 0): java.lang.OutOfMemoryError: Java heap space
	at java.util.Arrays.copyOf(Arrays.java:2271)
	at java.io.ByteArrayOutputStream.grow(ByteArrayOutputStream.java:118)
	at java.io.ByteArrayOutputStream.ensureCapacity(ByteArrayOutputStream.java:93)
	at java.io.ByteArrayOutputStream.write(ByteArrayOutputStream.java:153)
	at org.apache.spark.util.ByteBufferOutputStream.write(ByteBufferOutputStream.scala:41)
	at java.io.ObjectOutputStream$BlockDataOutputStream.write(ObjectOutputStream.java:1852)
	at java.io.ObjectOutputStream.write(ObjectOutputStream.java:708)
	at org.apache.spark.util.Utils$.writeByteBuffer(Utils.scala:238)
	at org.apache.spark.scheduler.DirectTaskResult$$anonfun$writeExternal$1.apply$mcV$sp(TaskResult.scala:50)
	at org.apache.spark.scheduler.DirectTaskResult$$anonfun$writeExternal$1.apply(TaskResult.scala:48)
	at org.apache.spark.scheduler.DirectTaskResult$$anonfun$writeExternal$1.apply(TaskResult.scala:48)
	at org.apache.spark.util.Utils$.tryOrIOException(Utils.scala:1269)
	at org.apache.spark.scheduler.DirectTaskResult.writeExternal(TaskResult.scala:48)
	at java.io.ObjectOutputStream.writeExternalData(ObjectOutputStream.java:1458)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1429)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1177)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:347)
	at org.apache.spark.serializer.JavaSerializationStream.writeObject(JavaSerializer.scala:43)
	at org.apache.spark.serializer.JavaSerializerInstance.serialize(JavaSerializer.scala:100)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:384)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

17/03/15 23:22:57 ERROR TaskSchedulerImpl: Lost executor 0 on 172.21.15.173: Remote RPC client disassociated. Likely due to containers exceeding thresholds, or network issues. Check driver logs for WARN messages.
17/03/15 23:22:57 ERROR TransportResponseHandler: Still have 1 requests outstanding when connection from /172.21.15.173:56667 is closed
17/03/15 23:22:57 WARN TaskSetManager: Lost task 0.1 in stage 5.0 (TID 26, 172.21.15.173, executor 0): ExecutorLostFailure (executor 0 exited caused by one of the running tasks) Reason: Remote RPC client disassociated. Likely due to containers exceeding thresholds, or network issues. Check driver logs for WARN messages.
17/03/15 23:22:57 WARN TaskSetManager: Lost task 1.0 in stage 5.0 (TID 25, 172.21.15.173, executor 0): ExecutorLostFailure (executor 0 exited caused by one of the running tasks) Reason: Remote RPC client disassociated. Likely due to containers exceeding thresholds, or network issues. Check driver logs for WARN messages.
17/03/15 23:22:57 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=13, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=22528
17/03/15 23:22:57 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=27941 lastFlushOffset=22583 createNewBlock=false
17/03/15 23:22:57 DEBUG DFSClient: Queued packet 13
17/03/15 23:22:57 DEBUG DFSClient: Waiting for ack for: 13
17/03/15 23:22:57 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809232_68409 sending packet packet seqno: 13 offsetInBlock: 22528 lastPacketInBlock: false lastByteOffsetInBlock: 27941
17/03/15 23:22:57 DEBUG DFSClient: DFSClient seqno: 13 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0
17/03/15 23:22:57 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=14, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=27648
17/03/15 23:22:57 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=27941 lastFlushOffset=27941 createNewBlock=false
17/03/15 23:22:57 DEBUG DFSClient: Waiting for ack for: 13
[Stage 5:>                                                          (0 + 0) / 2]17/03/15 23:22:58 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=15, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=27648
17/03/15 23:22:58 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=27941 lastFlushOffset=27941 createNewBlock=false
17/03/15 23:22:58 DEBUG DFSClient: Waiting for ack for: 13
17/03/15 23:22:58 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=16, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=27648
17/03/15 23:22:58 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=27941 lastFlushOffset=27941 createNewBlock=false
17/03/15 23:22:58 DEBUG DFSClient: Waiting for ack for: 13
[Stage 5:>                                                          (0 + 2) / 2]17/03/15 23:22:59 WARN TaskSetManager: Lost task 0.2 in stage 5.0 (TID 28, 172.21.15.173, executor 1): FetchFailed(null, shuffleId=1, mapId=-1, reduceId=0, message=
org.apache.spark.shuffle.MetadataFetchFailedException: Missing an output location for shuffle 1
	at org.apache.spark.MapOutputTracker$$anonfun$org$apache$spark$MapOutputTracker$$convertMapStatuses$2.apply(MapOutputTracker.scala:697)
	at org.apache.spark.MapOutputTracker$$anonfun$org$apache$spark$MapOutputTracker$$convertMapStatuses$2.apply(MapOutputTracker.scala:693)
	at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:733)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
	at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:732)
	at org.apache.spark.MapOutputTracker$.org$apache$spark$MapOutputTracker$$convertMapStatuses(MapOutputTracker.scala:693)
	at org.apache.spark.MapOutputTracker.getMapSizesByExecutorId(MapOutputTracker.scala:147)
	at org.apache.spark.shuffle.BlockStoreShuffleReader.read(BlockStoreShuffleReader.scala:49)
	at org.apache.spark.rdd.ShuffledRDD.compute(ShuffledRDD.scala:105)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:88)
	at org.apache.spark.scheduler.Task.run(Task.scala:114)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:322)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

)
17/03/15 23:22:59 WARN TaskSetManager: Lost task 1.1 in stage 5.0 (TID 27, 172.21.15.173, executor 1): FetchFailed(null, shuffleId=1, mapId=-1, reduceId=1, message=
org.apache.spark.shuffle.MetadataFetchFailedException: Missing an output location for shuffle 1
	at org.apache.spark.MapOutputTracker$$anonfun$org$apache$spark$MapOutputTracker$$convertMapStatuses$2.apply(MapOutputTracker.scala:697)
	at org.apache.spark.MapOutputTracker$$anonfun$org$apache$spark$MapOutputTracker$$convertMapStatuses$2.apply(MapOutputTracker.scala:693)
	at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:733)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
	at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:732)
	at org.apache.spark.MapOutputTracker$.org$apache$spark$MapOutputTracker$$convertMapStatuses(MapOutputTracker.scala:693)
	at org.apache.spark.MapOutputTracker.getMapSizesByExecutorId(MapOutputTracker.scala:147)
	at org.apache.spark.shuffle.BlockStoreShuffleReader.read(BlockStoreShuffleReader.scala:49)
	at org.apache.spark.rdd.ShuffledRDD.compute(ShuffledRDD.scala:105)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:88)
	at org.apache.spark.scheduler.Task.run(Task.scala:114)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:322)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

)
17/03/15 23:22:59 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=17, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=27648
17/03/15 23:22:59 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=27941 lastFlushOffset=27941 createNewBlock=false
17/03/15 23:22:59 DEBUG DFSClient: Waiting for ack for: 13
[Stage 4:>                                                         (0 + 4) / 10]17/03/15 23:23:02 ERROR RetryingBlockFetcher: Exception while beginning fetch of 1 outstanding blocks (after 1 retries)
java.io.IOException: Failed to connect to /172.21.15.173:56667
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:745)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:56667
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/15 23:23:07 ERROR RetryingBlockFetcher: Exception while beginning fetch of 1 outstanding blocks (after 2 retries)
java.io.IOException: Failed to connect to /172.21.15.173:56667
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:745)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:56667
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/15 23:23:07 DEBUG Client: The ping interval is 60000 ms.
17/03/15 23:23:07 DEBUG Client: Connecting to spark1/172.21.15.90:9000
17/03/15 23:23:07 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: starting, having connections 1
17/03/15 23:23:07 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #10
17/03/15 23:23:07 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #10
17/03/15 23:23:07 DEBUG ProtobufRpcEngine: Call: renewLease took 6ms
17/03/15 23:23:07 DEBUG LeaseRenewer: Lease renewed for client DFSClient_NONMAPREDUCE_-308821755_1
17/03/15 23:23:07 DEBUG LeaseRenewer: Lease renewer daemon for [DFSClient_NONMAPREDUCE_-308821755_1] with renew id 1 executed
17/03/15 23:23:12 ERROR RetryingBlockFetcher: Exception while beginning fetch of 1 outstanding blocks (after 3 retries)
java.io.IOException: Failed to connect to /172.21.15.173:56667
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:745)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:56667
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/15 23:23:12 WARN BlockManager: Failed to fetch block after 1 fetch failures. Most recent failure cause:
org.apache.spark.SparkException: Exception thrown in awaitResult: 
	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:205)
	at org.apache.spark.network.BlockTransferService.fetchBlockSync(BlockTransferService.scala:104)
	at org.apache.spark.storage.BlockManager.getRemoteBytes(BlockManager.scala:579)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply$mcV$sp(TaskResultGetter.scala:82)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply(TaskResultGetter.scala:63)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply(TaskResultGetter.scala:63)
	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:1951)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3.run(TaskResultGetter.scala:62)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.IOException: Failed to connect to /172.21.15.173:56667
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	... 1 more
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:56667
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/15 23:23:17 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: closed
17/03/15 23:23:17 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: stopped, remaining connections 0
[Stage 4:=================>                                        (3 + 4) / 10][Stage 4:=======================>                                  (4 + 4) / 10]17/03/15 23:23:37 DEBUG Client: The ping interval is 60000 ms.
17/03/15 23:23:37 DEBUG Client: Connecting to spark1/172.21.15.90:9000
17/03/15 23:23:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: starting, having connections 1
17/03/15 23:23:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #11
17/03/15 23:23:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #11
17/03/15 23:23:37 DEBUG ProtobufRpcEngine: Call: renewLease took 6ms
17/03/15 23:23:37 DEBUG LeaseRenewer: Lease renewed for client DFSClient_NONMAPREDUCE_-308821755_1
17/03/15 23:23:37 DEBUG LeaseRenewer: Lease renewer daemon for [DFSClient_NONMAPREDUCE_-308821755_1] with renew id 1 executed
17/03/15 23:23:47 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: closed
17/03/15 23:23:47 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: stopped, remaining connections 0
[Stage 4:========================================>                 (7 + 3) / 10][Stage 4:==============================================>           (8 + 2) / 10]17/03/15 23:24:07 DEBUG Client: The ping interval is 60000 ms.
17/03/15 23:24:07 DEBUG Client: Connecting to spark1/172.21.15.90:9000
17/03/15 23:24:07 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: starting, having connections 1
17/03/15 23:24:07 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #12
17/03/15 23:24:07 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #12
17/03/15 23:24:07 DEBUG ProtobufRpcEngine: Call: renewLease took 6ms
17/03/15 23:24:07 DEBUG LeaseRenewer: Lease renewed for client DFSClient_NONMAPREDUCE_-308821755_1
17/03/15 23:24:07 DEBUG LeaseRenewer: Lease renewer daemon for [DFSClient_NONMAPREDUCE_-308821755_1] with renew id 1 executed
17/03/15 23:24:15 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=18, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=27648
17/03/15 23:24:15 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=37496 lastFlushOffset=27941 createNewBlock=false
17/03/15 23:24:15 DEBUG DFSClient: Queued packet 18
17/03/15 23:24:15 DEBUG DFSClient: Waiting for ack for: 18
17/03/15 23:24:15 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809232_68409 sending packet packet seqno: 18 offsetInBlock: 27648 lastPacketInBlock: false lastByteOffsetInBlock: 37496
17/03/15 23:24:15 WARN DFSClient: Slow ReadProcessor read fields took 78033ms (threshold=30000ms); ack: seqno: 18 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0, targets: [DatanodeInfoWithStorage[172.21.15.173:50010,DS-bcd3842f-7acc-473a-9a24-360950418375,DISK]]
[Stage 5:>                                                          (0 + 2) / 2]17/03/15 23:24:16 WARN TaskSetManager: Lost task 1.0 in stage 5.1 (TID 40, 172.21.15.173, executor 1): java.lang.OutOfMemoryError: Java heap space
	at com.esotericsoftware.kryo.io.Output.require(Output.java:168)
	at com.esotericsoftware.kryo.io.Output.writeLong(Output.java:519)
	at com.esotericsoftware.kryo.io.Output.writeDouble(Output.java:648)
	at com.esotericsoftware.kryo.io.Output.writeDoubles(Output.java:729)
	at com.esotericsoftware.kryo.serializers.DefaultArraySerializers$DoubleArraySerializer.write(DefaultArraySerializers.java:216)
	at com.esotericsoftware.kryo.serializers.DefaultArraySerializers$DoubleArraySerializer.write(DefaultArraySerializers.java:205)
	at com.esotericsoftware.kryo.Kryo.writeObjectOrNull(Kryo.java:606)
	at com.esotericsoftware.kryo.serializers.ObjectField.write(ObjectField.java:87)
	at com.esotericsoftware.kryo.serializers.FieldSerializer.write(FieldSerializer.java:518)
	at com.esotericsoftware.kryo.Kryo.writeClassAndObject(Kryo.java:628)
	at com.twitter.chill.SomeSerializer.write(SomeSerializer.scala:21)
	at com.twitter.chill.SomeSerializer.write(SomeSerializer.scala:19)
	at com.esotericsoftware.kryo.Kryo.writeClassAndObject(Kryo.java:628)
	at org.apache.spark.serializer.KryoSerializerInstance.serialize(KryoSerializer.scala:312)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:364)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

17/03/15 23:24:17 ERROR TransportResponseHandler: Still have 1 requests outstanding when connection from /172.21.15.173:45645 is closed
17/03/15 23:24:17 ERROR TaskSchedulerImpl: Lost executor 1 on 172.21.15.173: Remote RPC client disassociated. Likely due to containers exceeding thresholds, or network issues. Check driver logs for WARN messages.
17/03/15 23:24:17 ERROR OneForOneBlockFetcher: Failed while starting block fetches
java.io.IOException: Connection from /172.21.15.173:45645 closed
	at org.apache.spark.network.client.TransportResponseHandler.channelInactive(TransportResponseHandler.java:128)
	at org.apache.spark.network.server.TransportChannelHandler.channelInactive(TransportChannelHandler.java:109)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:251)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:237)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:230)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelInactive(ChannelInboundHandlerAdapter.java:75)
	at io.netty.handler.timeout.IdleStateHandler.channelInactive(IdleStateHandler.java:257)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:251)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:237)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:230)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelInactive(ChannelInboundHandlerAdapter.java:75)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:251)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:237)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:230)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelInactive(ChannelInboundHandlerAdapter.java:75)
	at org.apache.spark.network.util.TransportFrameDecoder.channelInactive(TransportFrameDecoder.java:182)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:251)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:237)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:230)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelInactive(DefaultChannelPipeline.java:1289)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:251)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:237)
	at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:893)
	at io.netty.channel.AbstractChannel$AbstractUnsafe$7.run(AbstractChannel.java:691)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:408)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:455)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:745)
17/03/15 23:24:17 WARN TaskSetManager: Lost task 1.1 in stage 5.1 (TID 41, 172.21.15.173, executor 1): ExecutorLostFailure (executor 1 exited caused by one of the running tasks) Reason: Remote RPC client disassociated. Likely due to containers exceeding thresholds, or network issues. Check driver logs for WARN messages.
17/03/15 23:24:17 WARN TaskSetManager: Lost task 0.0 in stage 5.1 (TID 39, 172.21.15.173, executor 1): ExecutorLostFailure (executor 1 exited caused by one of the running tasks) Reason: Remote RPC client disassociated. Likely due to containers exceeding thresholds, or network issues. Check driver logs for WARN messages.
17/03/15 23:24:17 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=19, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=37376
17/03/15 23:24:17 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=41887 lastFlushOffset=37496 createNewBlock=false
17/03/15 23:24:17 DEBUG DFSClient: Queued packet 19
17/03/15 23:24:17 DEBUG DFSClient: Waiting for ack for: 19
17/03/15 23:24:17 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809232_68409 sending packet packet seqno: 19 offsetInBlock: 37376 lastPacketInBlock: false lastByteOffsetInBlock: 41887
17/03/15 23:24:17 DEBUG DFSClient: DFSClient seqno: 19 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0
17/03/15 23:24:17 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=20, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=41472
17/03/15 23:24:17 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=41887 lastFlushOffset=41887 createNewBlock=false
17/03/15 23:24:17 DEBUG DFSClient: Waiting for ack for: 19
[Stage 5:>                                                          (0 + 0) / 2]17/03/15 23:24:17 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: closed
17/03/15 23:24:17 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: stopped, remaining connections 0
17/03/15 23:24:18 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=21, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=41472
17/03/15 23:24:18 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=41887 lastFlushOffset=41887 createNewBlock=false
17/03/15 23:24:18 DEBUG DFSClient: Waiting for ack for: 19
17/03/15 23:24:18 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=22, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=41472
17/03/15 23:24:18 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=41887 lastFlushOffset=41887 createNewBlock=false
17/03/15 23:24:18 DEBUG DFSClient: Waiting for ack for: 19
[Stage 5:>                                                          (0 + 2) / 2]17/03/15 23:24:18 WARN TaskSetManager: Lost task 0.1 in stage 5.1 (TID 42, 172.21.15.173, executor 2): FetchFailed(null, shuffleId=1, mapId=-1, reduceId=0, message=
org.apache.spark.shuffle.MetadataFetchFailedException: Missing an output location for shuffle 1
	at org.apache.spark.MapOutputTracker$$anonfun$org$apache$spark$MapOutputTracker$$convertMapStatuses$2.apply(MapOutputTracker.scala:697)
	at org.apache.spark.MapOutputTracker$$anonfun$org$apache$spark$MapOutputTracker$$convertMapStatuses$2.apply(MapOutputTracker.scala:693)
	at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:733)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
	at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:732)
	at org.apache.spark.MapOutputTracker$.org$apache$spark$MapOutputTracker$$convertMapStatuses(MapOutputTracker.scala:693)
	at org.apache.spark.MapOutputTracker.getMapSizesByExecutorId(MapOutputTracker.scala:147)
	at org.apache.spark.shuffle.BlockStoreShuffleReader.read(BlockStoreShuffleReader.scala:49)
	at org.apache.spark.rdd.ShuffledRDD.compute(ShuffledRDD.scala:105)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:88)
	at org.apache.spark.scheduler.Task.run(Task.scala:114)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:322)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

)
17/03/15 23:24:18 WARN TaskSetManager: Lost task 1.2 in stage 5.1 (TID 43, 172.21.15.173, executor 2): FetchFailed(null, shuffleId=1, mapId=-1, reduceId=1, message=
org.apache.spark.shuffle.MetadataFetchFailedException: Missing an output location for shuffle 1
	at org.apache.spark.MapOutputTracker$$anonfun$org$apache$spark$MapOutputTracker$$convertMapStatuses$2.apply(MapOutputTracker.scala:697)
	at org.apache.spark.MapOutputTracker$$anonfun$org$apache$spark$MapOutputTracker$$convertMapStatuses$2.apply(MapOutputTracker.scala:693)
	at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:733)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
	at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:732)
	at org.apache.spark.MapOutputTracker$.org$apache$spark$MapOutputTracker$$convertMapStatuses(MapOutputTracker.scala:693)
	at org.apache.spark.MapOutputTracker.getMapSizesByExecutorId(MapOutputTracker.scala:147)
	at org.apache.spark.shuffle.BlockStoreShuffleReader.read(BlockStoreShuffleReader.scala:49)
	at org.apache.spark.rdd.ShuffledRDD.compute(ShuffledRDD.scala:105)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:88)
	at org.apache.spark.scheduler.Task.run(Task.scala:114)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:322)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

)
17/03/15 23:24:18 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=23, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=41472
17/03/15 23:24:18 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=41887 lastFlushOffset=41887 createNewBlock=false
17/03/15 23:24:18 DEBUG DFSClient: Waiting for ack for: 19
[Stage 4:>                                                         (0 + 4) / 10]17/03/15 23:24:22 ERROR RetryingBlockFetcher: Exception while beginning fetch of 1 outstanding blocks (after 1 retries)
java.io.IOException: Failed to connect to /172.21.15.173:45645
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:745)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:45645
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/15 23:24:27 ERROR RetryingBlockFetcher: Exception while beginning fetch of 1 outstanding blocks (after 2 retries)
java.io.IOException: Failed to connect to /172.21.15.173:45645
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:745)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:45645
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/15 23:24:32 ERROR RetryingBlockFetcher: Exception while beginning fetch of 1 outstanding blocks (after 3 retries)
java.io.IOException: Failed to connect to /172.21.15.173:45645
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:745)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:45645
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/15 23:24:32 WARN BlockManager: Failed to fetch block after 1 fetch failures. Most recent failure cause:
org.apache.spark.SparkException: Exception thrown in awaitResult: 
	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:205)
	at org.apache.spark.network.BlockTransferService.fetchBlockSync(BlockTransferService.scala:104)
	at org.apache.spark.storage.BlockManager.getRemoteBytes(BlockManager.scala:579)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply$mcV$sp(TaskResultGetter.scala:82)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply(TaskResultGetter.scala:63)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply(TaskResultGetter.scala:63)
	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:1951)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3.run(TaskResultGetter.scala:62)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.IOException: Failed to connect to /172.21.15.173:45645
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	... 1 more
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:45645
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/15 23:24:37 DEBUG Client: The ping interval is 60000 ms.
17/03/15 23:24:37 DEBUG Client: Connecting to spark1/172.21.15.90:9000
17/03/15 23:24:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: starting, having connections 1
17/03/15 23:24:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #13
17/03/15 23:24:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #13
17/03/15 23:24:37 DEBUG ProtobufRpcEngine: Call: renewLease took 7ms
17/03/15 23:24:37 DEBUG LeaseRenewer: Lease renewed for client DFSClient_NONMAPREDUCE_-308821755_1
17/03/15 23:24:37 DEBUG LeaseRenewer: Lease renewer daemon for [DFSClient_NONMAPREDUCE_-308821755_1] with renew id 1 executed
17/03/15 23:24:47 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: closed
17/03/15 23:24:47 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: stopped, remaining connections 0
[Stage 4:=======================>                                  (4 + 4) / 10]17/03/15 23:25:07 DEBUG Client: The ping interval is 60000 ms.
17/03/15 23:25:07 DEBUG Client: Connecting to spark1/172.21.15.90:9000
17/03/15 23:25:07 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: starting, having connections 1
17/03/15 23:25:07 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #14
17/03/15 23:25:07 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #14
17/03/15 23:25:07 DEBUG ProtobufRpcEngine: Call: renewLease took 6ms
17/03/15 23:25:07 DEBUG LeaseRenewer: Lease renewed for client DFSClient_NONMAPREDUCE_-308821755_1
17/03/15 23:25:07 DEBUG LeaseRenewer: Lease renewer daemon for [DFSClient_NONMAPREDUCE_-308821755_1] with renew id 1 executed
17/03/15 23:25:17 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: closed
17/03/15 23:25:17 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: stopped, remaining connections 0
[Stage 4:=============================>                            (5 + 4) / 10][Stage 4:==============================================>           (8 + 2) / 10]17/03/15 23:25:34 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=24, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=41472
17/03/15 23:25:34 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=51387 lastFlushOffset=41887 createNewBlock=false
17/03/15 23:25:34 DEBUG DFSClient: Queued packet 24
17/03/15 23:25:34 DEBUG DFSClient: Waiting for ack for: 24
17/03/15 23:25:34 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809232_68409 sending packet packet seqno: 24 offsetInBlock: 41472 lastPacketInBlock: false lastByteOffsetInBlock: 51387
17/03/15 23:25:34 WARN DFSClient: Slow ReadProcessor read fields took 77560ms (threshold=30000ms); ack: seqno: 24 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0, targets: [DatanodeInfoWithStorage[172.21.15.173:50010,DS-bcd3842f-7acc-473a-9a24-360950418375,DISK]]
[Stage 5:>                                                          (0 + 2) / 2]17/03/15 23:25:36 WARN TaskSetManager: Lost task 0.0 in stage 5.2 (TID 54, 172.21.15.173, executor 2): java.lang.OutOfMemoryError: Java heap space
	at java.util.Arrays.copyOf(Arrays.java:2271)
	at java.io.ByteArrayOutputStream.grow(ByteArrayOutputStream.java:118)
	at java.io.ByteArrayOutputStream.ensureCapacity(ByteArrayOutputStream.java:93)
	at java.io.ByteArrayOutputStream.write(ByteArrayOutputStream.java:153)
	at org.apache.spark.util.ByteBufferOutputStream.write(ByteBufferOutputStream.scala:41)
	at java.io.ObjectOutputStream$BlockDataOutputStream.write(ObjectOutputStream.java:1852)
	at java.io.ObjectOutputStream.write(ObjectOutputStream.java:708)
	at org.apache.spark.util.Utils$.writeByteBuffer(Utils.scala:238)
	at org.apache.spark.scheduler.DirectTaskResult$$anonfun$writeExternal$1.apply$mcV$sp(TaskResult.scala:50)
	at org.apache.spark.scheduler.DirectTaskResult$$anonfun$writeExternal$1.apply(TaskResult.scala:48)
	at org.apache.spark.scheduler.DirectTaskResult$$anonfun$writeExternal$1.apply(TaskResult.scala:48)
	at org.apache.spark.util.Utils$.tryOrIOException(Utils.scala:1269)
	at org.apache.spark.scheduler.DirectTaskResult.writeExternal(TaskResult.scala:48)
	at java.io.ObjectOutputStream.writeExternalData(ObjectOutputStream.java:1458)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1429)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1177)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:347)
	at org.apache.spark.serializer.JavaSerializationStream.writeObject(JavaSerializer.scala:43)
	at org.apache.spark.serializer.JavaSerializerInstance.serialize(JavaSerializer.scala:100)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:384)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

[Stage 5:>                                                          (0 + 1) / 2]17/03/15 23:25:36 WARN TaskSetManager: Lost task 0.1 in stage 5.2 (TID 56, 172.21.15.173, executor 2): FetchFailed(BlockManagerId(2, 172.21.15.173, 33403, None), shuffleId=1, mapId=0, reduceId=0, message=
org.apache.spark.shuffle.FetchFailedException: Error in opening FileSegmentManagedBuffer{file=/tmp/spark-cc7da6f7-b4cb-4f06-a1d6-6ba5deb00e97/executor-d7506859-a203-4613-a51e-919ed1c39a10/blockmgr-5e85c033-adc9-47c9-bbc5-83aa81238e0a/2b/shuffle_1_0_0.data, offset=0, length=36035171}
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.throwFetchFailedException(ShuffleBlockFetcherIterator.scala:416)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:356)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:58)
	at scala.collection.Iterator$$anon$12.nextCur(Iterator.scala:434)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:440)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at org.apache.spark.util.CompletionIterator.hasNext(CompletionIterator.scala:32)
	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:39)
	at org.apache.spark.util.collection.ExternalAppendOnlyMap.insertAll(ExternalAppendOnlyMap.scala:154)
	at org.apache.spark.Aggregator.combineCombinersByKey(Aggregator.scala:50)
	at org.apache.spark.shuffle.BlockStoreShuffleReader.read(BlockStoreShuffleReader.scala:82)
	at org.apache.spark.rdd.ShuffledRDD.compute(ShuffledRDD.scala:105)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:88)
	at org.apache.spark.scheduler.Task.run(Task.scala:114)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:322)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.IOException: Error in opening FileSegmentManagedBuffer{file=/tmp/spark-cc7da6f7-b4cb-4f06-a1d6-6ba5deb00e97/executor-d7506859-a203-4613-a51e-919ed1c39a10/blockmgr-5e85c033-adc9-47c9-bbc5-83aa81238e0a/2b/shuffle_1_0_0.data, offset=0, length=36035171}
	at org.apache.spark.network.buffer.FileSegmentManagedBuffer.createInputStream(FileSegmentManagedBuffer.java:113)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:349)
	... 21 more
Caused by: java.io.FileNotFoundException: /tmp/spark-cc7da6f7-b4cb-4f06-a1d6-6ba5deb00e97/executor-d7506859-a203-4613-a51e-919ed1c39a10/blockmgr-5e85c033-adc9-47c9-bbc5-83aa81238e0a/2b/shuffle_1_0_0.data (没有那个文件或目录)
	at java.io.FileInputStream.open(Native Method)
	at java.io.FileInputStream.<init>(FileInputStream.java:146)
	at org.apache.spark.network.buffer.FileSegmentManagedBuffer.createInputStream(FileSegmentManagedBuffer.java:98)
	... 22 more

)
17/03/15 23:25:36 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=25, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=51200
17/03/15 23:25:36 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=56596 lastFlushOffset=51387 createNewBlock=false
17/03/15 23:25:36 DEBUG DFSClient: Queued packet 25
17/03/15 23:25:36 DEBUG DFSClient: Waiting for ack for: 25
17/03/15 23:25:36 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809232_68409 sending packet packet seqno: 25 offsetInBlock: 51200 lastPacketInBlock: false lastByteOffsetInBlock: 56596
17/03/15 23:25:36 DEBUG DFSClient: DFSClient seqno: 25 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0
17/03/15 23:25:36 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=26, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=56320
17/03/15 23:25:36 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=56596 lastFlushOffset=56596 createNewBlock=false
17/03/15 23:25:36 DEBUG DFSClient: Waiting for ack for: 25
[Stage 4:>                                                         (0 + 0) / 10]17/03/15 23:25:36 ERROR TaskSchedulerImpl: Lost executor 2 on 172.21.15.173: Remote RPC client disassociated. Likely due to containers exceeding thresholds, or network issues. Check driver logs for WARN messages.
17/03/15 23:25:36 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=27, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=56320
17/03/15 23:25:36 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=56596 lastFlushOffset=56596 createNewBlock=false
17/03/15 23:25:36 DEBUG DFSClient: Waiting for ack for: 25
17/03/15 23:25:36 ERROR TransportResponseHandler: Still have 1 requests outstanding when connection from /172.21.15.173:33403 is closed
17/03/15 23:25:37 DEBUG Client: The ping interval is 60000 ms.
17/03/15 23:25:37 DEBUG Client: Connecting to spark1/172.21.15.90:9000
17/03/15 23:25:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: starting, having connections 1
17/03/15 23:25:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #15
17/03/15 23:25:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #15
17/03/15 23:25:37 DEBUG ProtobufRpcEngine: Call: renewLease took 8ms
17/03/15 23:25:37 DEBUG LeaseRenewer: Lease renewed for client DFSClient_NONMAPREDUCE_-308821755_1
17/03/15 23:25:37 DEBUG LeaseRenewer: Lease renewer daemon for [DFSClient_NONMAPREDUCE_-308821755_1] with renew id 1 executed
17/03/15 23:25:38 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=28, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=56320
17/03/15 23:25:38 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=56596 lastFlushOffset=56596 createNewBlock=false
17/03/15 23:25:38 DEBUG DFSClient: Waiting for ack for: 25
17/03/15 23:25:38 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=29, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=56320
17/03/15 23:25:38 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=56596 lastFlushOffset=56596 createNewBlock=false
17/03/15 23:25:38 DEBUG DFSClient: Waiting for ack for: 25
[Stage 4:>                                                         (0 + 4) / 10]17/03/15 23:25:41 ERROR RetryingBlockFetcher: Exception while beginning fetch of 1 outstanding blocks (after 1 retries)
java.io.IOException: Failed to connect to /172.21.15.173:33403
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:745)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:33403
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/15 23:25:46 ERROR RetryingBlockFetcher: Exception while beginning fetch of 1 outstanding blocks (after 2 retries)
java.io.IOException: Failed to connect to /172.21.15.173:33403
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:745)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:33403
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/15 23:25:47 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: closed
17/03/15 23:25:47 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: stopped, remaining connections 0
17/03/15 23:25:51 ERROR RetryingBlockFetcher: Exception while beginning fetch of 1 outstanding blocks (after 3 retries)
java.io.IOException: Failed to connect to /172.21.15.173:33403
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:745)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:33403
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/15 23:25:51 WARN BlockManager: Failed to fetch block after 1 fetch failures. Most recent failure cause:
org.apache.spark.SparkException: Exception thrown in awaitResult: 
	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:205)
	at org.apache.spark.network.BlockTransferService.fetchBlockSync(BlockTransferService.scala:104)
	at org.apache.spark.storage.BlockManager.getRemoteBytes(BlockManager.scala:579)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply$mcV$sp(TaskResultGetter.scala:82)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply(TaskResultGetter.scala:63)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply(TaskResultGetter.scala:63)
	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:1951)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3.run(TaskResultGetter.scala:62)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.IOException: Failed to connect to /172.21.15.173:33403
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	... 1 more
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:33403
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/15 23:25:51 WARN TaskSetManager: Lost task 1.0 in stage 5.2 (TID 55, 172.21.15.173, executor 2): TaskResultLost (result lost from block manager)
17/03/15 23:26:07 DEBUG Client: The ping interval is 60000 ms.
17/03/15 23:26:07 DEBUG Client: Connecting to spark1/172.21.15.90:9000
17/03/15 23:26:07 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: starting, having connections 1
17/03/15 23:26:07 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #16
17/03/15 23:26:07 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #16
17/03/15 23:26:07 DEBUG ProtobufRpcEngine: Call: renewLease took 6ms
17/03/15 23:26:07 DEBUG LeaseRenewer: Lease renewed for client DFSClient_NONMAPREDUCE_-308821755_1
17/03/15 23:26:07 DEBUG LeaseRenewer: Lease renewer daemon for [DFSClient_NONMAPREDUCE_-308821755_1] with renew id 1 executed
[Stage 4:=======================>                                  (4 + 4) / 10]17/03/15 23:26:17 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: closed
17/03/15 23:26:17 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: stopped, remaining connections 0
17/03/15 23:26:37 DEBUG Client: The ping interval is 60000 ms.
17/03/15 23:26:37 DEBUG Client: Connecting to spark1/172.21.15.90:9000
17/03/15 23:26:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: starting, having connections 1
17/03/15 23:26:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #17
17/03/15 23:26:37 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #17
17/03/15 23:26:37 DEBUG ProtobufRpcEngine: Call: renewLease took 5ms
17/03/15 23:26:37 DEBUG LeaseRenewer: Lease renewed for client DFSClient_NONMAPREDUCE_-308821755_1
17/03/15 23:26:37 DEBUG LeaseRenewer: Lease renewer daemon for [DFSClient_NONMAPREDUCE_-308821755_1] with renew id 1 executed
[Stage 4:=============================>                            (5 + 4) / 10][Stage 4:========================================>                 (7 + 3) / 10][Stage 4:==============================================>           (8 + 2) / 10]17/03/15 23:26:47 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: closed
17/03/15 23:26:47 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: stopped, remaining connections 0
[Stage 4:====================================================>     (9 + 1) / 10]17/03/15 23:26:55 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=30, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=56320
17/03/15 23:26:55 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=66013 lastFlushOffset=56596 createNewBlock=false
17/03/15 23:26:55 DEBUG DFSClient: Queued packet 30
17/03/15 23:26:55 DEBUG DFSClient: Waiting for ack for: 30
17/03/15 23:26:55 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809232_68409 sending packet packet seqno: 30 offsetInBlock: 56320 lastPacketInBlock: false lastByteOffsetInBlock: 66013
17/03/15 23:26:55 WARN DFSClient: Slow ReadProcessor read fields took 79530ms (threshold=30000ms); ack: seqno: 30 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0, targets: [DatanodeInfoWithStorage[172.21.15.173:50010,DS-bcd3842f-7acc-473a-9a24-360950418375,DISK]]
[Stage 5:>                                                          (0 + 2) / 2]17/03/15 23:26:57 WARN TaskSetManager: Lost task 0.0 in stage 5.3 (TID 67, 172.21.15.173, executor 3): java.lang.OutOfMemoryError: Java heap space
	at com.esotericsoftware.kryo.io.Output.require(Output.java:168)
	at com.esotericsoftware.kryo.io.Output.writeLong(Output.java:519)
	at com.esotericsoftware.kryo.io.Output.writeDouble(Output.java:648)
	at com.esotericsoftware.kryo.io.Output.writeDoubles(Output.java:729)
	at com.esotericsoftware.kryo.serializers.DefaultArraySerializers$DoubleArraySerializer.write(DefaultArraySerializers.java:216)
	at com.esotericsoftware.kryo.serializers.DefaultArraySerializers$DoubleArraySerializer.write(DefaultArraySerializers.java:205)
	at com.esotericsoftware.kryo.Kryo.writeObjectOrNull(Kryo.java:606)
	at com.esotericsoftware.kryo.serializers.ObjectField.write(ObjectField.java:87)
	at com.esotericsoftware.kryo.serializers.FieldSerializer.write(FieldSerializer.java:518)
	at com.esotericsoftware.kryo.Kryo.writeClassAndObject(Kryo.java:628)
	at com.twitter.chill.SomeSerializer.write(SomeSerializer.scala:21)
	at com.twitter.chill.SomeSerializer.write(SomeSerializer.scala:19)
	at com.esotericsoftware.kryo.Kryo.writeClassAndObject(Kryo.java:628)
	at org.apache.spark.serializer.KryoSerializerInstance.serialize(KryoSerializer.scala:312)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:364)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

17/03/15 23:26:57 WARN TaskSetManager: Lost task 0.1 in stage 5.3 (TID 69, 172.21.15.173, executor 3): FetchFailed(BlockManagerId(3, 172.21.15.173, 46056, None), shuffleId=1, mapId=0, reduceId=0, message=
org.apache.spark.shuffle.FetchFailedException: Error in opening FileSegmentManagedBuffer{file=/tmp/spark-cc7da6f7-b4cb-4f06-a1d6-6ba5deb00e97/executor-d7506859-a203-4613-a51e-919ed1c39a10/blockmgr-db9faa1d-a097-46ca-b301-bb67ed3b9569/2b/shuffle_1_0_0.data, offset=0, length=36035171}
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.throwFetchFailedException(ShuffleBlockFetcherIterator.scala:416)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:356)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:58)
	at scala.collection.Iterator$$anon$12.nextCur(Iterator.scala:434)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:440)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at org.apache.spark.util.CompletionIterator.hasNext(CompletionIterator.scala:32)
	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:39)
	at org.apache.spark.util.collection.ExternalAppendOnlyMap.insertAll(ExternalAppendOnlyMap.scala:154)
	at org.apache.spark.Aggregator.combineCombinersByKey(Aggregator.scala:50)
	at org.apache.spark.shuffle.BlockStoreShuffleReader.read(BlockStoreShuffleReader.scala:82)
	at org.apache.spark.rdd.ShuffledRDD.compute(ShuffledRDD.scala:105)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:88)
	at org.apache.spark.scheduler.Task.run(Task.scala:114)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:322)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.IOException: Error in opening FileSegmentManagedBuffer{file=/tmp/spark-cc7da6f7-b4cb-4f06-a1d6-6ba5deb00e97/executor-d7506859-a203-4613-a51e-919ed1c39a10/blockmgr-db9faa1d-a097-46ca-b301-bb67ed3b9569/2b/shuffle_1_0_0.data, offset=0, length=36035171}
	at org.apache.spark.network.buffer.FileSegmentManagedBuffer.createInputStream(FileSegmentManagedBuffer.java:113)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:349)
	... 21 more
Caused by: java.io.FileNotFoundException: /tmp/spark-cc7da6f7-b4cb-4f06-a1d6-6ba5deb00e97/executor-d7506859-a203-4613-a51e-919ed1c39a10/blockmgr-db9faa1d-a097-46ca-b301-bb67ed3b9569/2b/shuffle_1_0_0.data (没有那个文件或目录)
	at java.io.FileInputStream.open(Native Method)
	at java.io.FileInputStream.<init>(FileInputStream.java:146)
	at org.apache.spark.network.buffer.FileSegmentManagedBuffer.createInputStream(FileSegmentManagedBuffer.java:98)
	... 22 more

)
17/03/15 23:26:57 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=31, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=65536
17/03/15 23:26:57 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=71806 lastFlushOffset=66013 createNewBlock=false
17/03/15 23:26:57 DEBUG DFSClient: Queued packet 31
17/03/15 23:26:57 DEBUG DFSClient: Waiting for ack for: 31
17/03/15 23:26:57 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809232_68409 sending packet packet seqno: 31 offsetInBlock: 65536 lastPacketInBlock: false lastByteOffsetInBlock: 71806
17/03/15 23:26:57 DEBUG DFSClient: DFSClient seqno: 31 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0
Exception in thread "main" org.apache.spark.SparkException: Job aborted due to stage failure: ResultStage 5 (treeAggregate at RowMatrix.scala:122) has failed the maximum allowable number of times: 4. Most recent failure reason: org.apache.spark.shuffle.FetchFailedException: Error in opening FileSegmentManagedBuffer{file=/tmp/spark-cc7da6f7-b4cb-4f06-a1d6-6ba5deb00e97/executor-d7506859-a203-4613-a51e-919ed1c39a10/blockmgr-db9faa1d-a097-46ca-b301-bb67ed3b9569/2b/shuffle_1_0_0.data, offset=0, length=36035171} 	at org.apache.spark.storage.ShuffleBlockFetcherIterator.throwFetchFailedException(ShuffleBlockFetcherIterator.scala:416) 	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:356) 	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:58) 	at scala.collection.Iterator$$anon$12.nextCur(Iterator.scala:434) 	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:440) 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408) 	at org.apache.spark.util.CompletionIterator.hasNext(CompletionIterator.scala:32) 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:39) 	at org.apache.spark.util.collection.ExternalAppendOnlyMap.insertAll(ExternalAppendOnlyMap.scala:154) 	at org.apache.spark.Aggregator.combineCombinersByKey(Aggregator.scala:50) 	at org.apache.spark.shuffle.BlockStoreShuffleReader.read(BlockStoreShuffleReader.scala:82) 	at org.apache.spark.rdd.ShuffledRDD.compute(ShuffledRDD.scala:105) 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323) 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287) 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38) 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323) 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287) 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:88) 	at org.apache.spark.scheduler.Task.run(Task.scala:114) 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:322) 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) 	at java.lang.Thread.run(Thread.java:745) Caused by: java.io.IOException: Error in opening FileSegmentManagedBuffer{file=/tmp/spark-cc7da6f7-b4cb-4f06-a1d6-6ba5deb00e97/executor-d7506859-a203-4613-a51e-919ed1c39a10/blockmgr-db9faa1d-a097-46ca-b301-bb67ed3b9569/2b/shuffle_1_0_0.data, offset=0, length=36035171} 	at org.apache.spark.network.buffer.FileSegmentManagedBuffer.createInputStream(FileSegmentManagedBuffer.java:113) 	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:349) 	... 21 more Caused by: java.io.FileNotFoundException: /tmp/spark-cc7da6f7-b4cb-4f06-a1d6-6ba5deb00e97/executor-d7506859-a203-4613-a51e-919ed1c39a10/blockmgr-db9faa1d-a097-46ca-b301-bb67ed3b9569/2b/shuffle_1_0_0.data (没有那个文件或目录) 	at java.io.FileInputStream.open(Native Method) 	at java.io.FileInputStream.<init>(FileInputStream.java:146) 	at org.apache.spark.network.buffer.FileSegmentManagedBuffer.createInputStream(FileSegmentManagedBuffer.java:98) 	... 22 more 
	at org.apache.spark.scheduler.DAGScheduler.org$apache$spark$scheduler$DAGScheduler$$failJobAndIndependentStages(DAGScheduler.scala:1456)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1444)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1443)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1443)
	at org.apache.spark.scheduler.DAGScheduler.handleTaskCompletion(DAGScheduler.scala:1273)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:1668)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1626)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1615)
	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:48)
	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:628)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2016)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2113)
	at org.apache.spark.rdd.RDD$$anonfun$reduce$1.apply(RDD.scala:1027)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
	at org.apache.spark.rdd.RDD.withScope(RDD.scala:364)
	at org.apache.spark.rdd.RDD.reduce(RDD.scala:1009)
	at org.apache.spark.rdd.RDD$$anonfun$treeAggregate$1.apply(RDD.scala:1152)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
	at org.apache.spark.rdd.RDD.withScope(RDD.scala:364)
	at org.apache.spark.rdd.RDD.treeAggregate(RDD.scala:1129)
	at org.apache.spark.mllib.linalg.distributed.RowMatrix.computeGramianMatrix(RowMatrix.scala:122)
	at org.apache.spark.mllib.linalg.distributed.RowMatrix.computeCovariance(RowMatrix.scala:344)
	at org.apache.spark.mllib.linalg.distributed.RowMatrix.computePrincipalComponentsAndExplainedVariance(RowMatrix.scala:387)
	at org.apache.spark.mllib.feature.PCA.fit(PCA.scala:48)
	at PCA.src.main.scala.PCAApp$.main(PCAApp.scala:58)
	at PCA.src.main.scala.PCAApp.main(PCAApp.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.spark.deploy.SparkSubmit$.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:728)
	at org.apache.spark.deploy.SparkSubmit$.doRunMain$1(SparkSubmit.scala:177)
	at org.apache.spark.deploy.SparkSubmit$.submit(SparkSubmit.scala:202)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:116)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
17/03/15 23:26:57 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=32, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=71680
17/03/15 23:26:57 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=71806 lastFlushOffset=71806 createNewBlock=false
17/03/15 23:26:57 DEBUG DFSClient: Waiting for ack for: 31
17/03/15 23:26:57 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=33, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=71680
17/03/15 23:26:57 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=71806 lastFlushOffset=71806 createNewBlock=false
17/03/15 23:26:57 DEBUG DFSClient: Waiting for ack for: 31
17/03/15 23:26:57 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=34, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=71680
17/03/15 23:26:57 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=71806 lastFlushOffset=71806 createNewBlock=false
17/03/15 23:26:57 DEBUG DFSClient: Waiting for ack for: 31
17/03/15 23:26:57 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=35, src=/eventLogs/app-20170315232137-0002.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=71680
17/03/15 23:26:57 DEBUG DFSClient: Queued packet 35
17/03/15 23:26:57 DEBUG DFSClient: Queued packet 36
17/03/15 23:26:57 DEBUG DFSClient: Waiting for ack for: 36
17/03/15 23:26:57 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809232_68409 sending packet packet seqno: 35 offsetInBlock: 71680 lastPacketInBlock: false lastByteOffsetInBlock: 76080
17/03/15 23:26:57 DEBUG DFSClient: DFSClient seqno: 35 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0
17/03/15 23:26:57 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809232_68409 sending packet packet seqno: 36 offsetInBlock: 76080 lastPacketInBlock: true lastByteOffsetInBlock: 76080
17/03/15 23:26:57 DEBUG DFSClient: DFSClient seqno: 36 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0
17/03/15 23:26:57 DEBUG DFSClient: Closing old block BP-519507147-172.21.15.90-1479901973323:blk_1073809232_68409
17/03/15 23:26:57 DEBUG Client: The ping interval is 60000 ms.
17/03/15 23:26:57 DEBUG Client: Connecting to spark1/172.21.15.90:9000
17/03/15 23:26:57 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: starting, having connections 1
17/03/15 23:26:57 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #18
17/03/15 23:26:57 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #18
17/03/15 23:26:57 DEBUG ProtobufRpcEngine: Call: complete took 15ms
17/03/15 23:26:57 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #19
17/03/15 23:26:57 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #19
17/03/15 23:26:57 DEBUG ProtobufRpcEngine: Call: getFileInfo took 3ms
17/03/15 23:26:57 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #20
17/03/15 23:26:57 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #20
17/03/15 23:26:57 DEBUG ProtobufRpcEngine: Call: rename took 7ms
17/03/15 23:26:57 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop sending #21
17/03/15 23:26:57 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop got value #21
17/03/15 23:26:57 DEBUG ProtobufRpcEngine: Call: setTimes took 9ms
17/03/15 23:26:57 ERROR Utils: Uncaught exception in thread task-result-getter-1
java.lang.InterruptedException
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:996)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1303)
	at scala.concurrent.impl.Promise$DefaultPromise.tryAwait(Promise.scala:202)
	at scala.concurrent.impl.Promise$DefaultPromise.ready(Promise.scala:218)
	at scala.concurrent.impl.Promise$DefaultPromise.result(Promise.scala:223)
	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:201)
	at org.apache.spark.network.BlockTransferService.fetchBlockSync(BlockTransferService.scala:104)
	at org.apache.spark.storage.BlockManager.getRemoteBytes(BlockManager.scala:579)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply$mcV$sp(TaskResultGetter.scala:82)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply(TaskResultGetter.scala:63)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply(TaskResultGetter.scala:63)
	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:1951)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3.run(TaskResultGetter.scala:62)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Exception in thread "task-result-getter-1" java.lang.Error: java.lang.InterruptedException
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1151)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.InterruptedException
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:996)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1303)
	at scala.concurrent.impl.Promise$DefaultPromise.tryAwait(Promise.scala:202)
	at scala.concurrent.impl.Promise$DefaultPromise.ready(Promise.scala:218)
	at scala.concurrent.impl.Promise$DefaultPromise.result(Promise.scala:223)
	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:201)
	at org.apache.spark.network.BlockTransferService.fetchBlockSync(BlockTransferService.scala:104)
	at org.apache.spark.storage.BlockManager.getRemoteBytes(BlockManager.scala:579)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply$mcV$sp(TaskResultGetter.scala:82)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply(TaskResultGetter.scala:63)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply(TaskResultGetter.scala:63)
	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:1951)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3.run(TaskResultGetter.scala:62)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	... 2 more
17/03/15 23:26:57 DEBUG PoolThreadCache: Freed 6 thread-local buffer(s) from thread: shuffle-server-6-4
17/03/15 23:26:57 DEBUG PoolThreadCache: Freed 11 thread-local buffer(s) from thread: shuffle-server-6-2
17/03/15 23:26:57 DEBUG PoolThreadCache: Freed 6 thread-local buffer(s) from thread: shuffle-server-6-3
17/03/15 23:26:57 ERROR TransportResponseHandler: Still have 1 requests outstanding when connection from /172.21.15.173:46056 is closed
17/03/15 23:26:57 DEBUG PoolThreadCache: Freed 19 thread-local buffer(s) from thread: rpc-server-3-2
17/03/15 23:26:57 DEBUG PoolThreadCache: Freed 66 thread-local buffer(s) from thread: rpc-server-3-3
17/03/15 23:26:57 DEBUG Client: stopping client from cache: org.apache.hadoop.ipc.Client@1a6f3919
17/03/15 23:26:57 DEBUG Client: removing client from cache: org.apache.hadoop.ipc.Client@1a6f3919
17/03/15 23:26:57 DEBUG Client: stopping actual client because no more references remain: org.apache.hadoop.ipc.Client@1a6f3919
17/03/15 23:26:57 DEBUG Client: Stopping client
17/03/15 23:26:57 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: closed
17/03/15 23:26:57 DEBUG Client: IPC Client (1886667974) connection to spark1/172.21.15.90:9000 from hadoop: stopped, remaining connections 0
